pilot/pkg/features/pilot.go

```diff
@@ -653,6 +653,9 @@ var (
 	MetricGracefulDeletionInterval = env.Register("METRIC_GRACEFUL_DELETION_INTERVAL", 5*time.Minute,
 		"Metric expiry graceful deletion interval. No-op if METRIC_ROTATION_INTERVAL is disabled.").Get()
 
+	NativeMetadataExchange = env.Register("NATIVE_METADATA_EXCHANGE", true,
```

Comment:
Should the default value of "NATIVE_METADATA_EXCHANGE" be false?  A user can opt in to try it by setting the flag to be true and when it becomes stable, the flag can be set as true by default.

Reply:
How would we stabilize it if it's off-by-default for new clusters? I think it's indeed some risk, but this filter is fairly simple that it might be OK to avoid a prolonged maintenance interval for the Wasm version.

Reply:
+1 to **on** by default. 

Off by default is good for cool new features that users will want to use, but have some risk associated such that we cannot turn them on.

On by default is good for improvements we make that we are worried about unknown-risks, so we have an escape hatch if something goes wrong.

off-by-default for something like this will not help us. No one will opt into an internal refactoring, so we will not get feedback.

even better, this is merging that the start of the release cycle so we have 3 months before it ships to users (unless we planned to backport to 1.19)

Reply:
> an internal refactoring,

+1 to on by default for this reason

Reply:
definitely +1 for 1.20. Worth considering for 1.19 as well if pass all of our testing.

Reply:
SGTM.